core/src/main/scala/org/apache/spark/ui/ToolTips.scala

```diff
@@ -62,6 +62,13 @@ private[spark] object ToolTips {
     """Time that the executor spent paused for Java garbage collection while the task was
        running."""
 
+  val PEAK_EXECUTION_MEMORY =
+    """Execution memory refers to the memory used by internal data structures created during
+       shuffles, aggregations and joins when Tungsten is enabled. The value of this accumulator
+       should be approximately the sum of the peak sizes across all such data structures created
+       in this task. For SQL jobs, this only tracks all unsafe operators, broadcast joins, and
+       external sort."""
```

Comment:
Yeah, I agree the inconsistency is the biggest problem. I actually think the best thing might be to rename "spark.shuffle.memoryFraction" to be "spark.execution.memoryFraction" so it's consistent with this header (we can still accept the old one for backwards compatibility).

Overall I feel that "execution" is a substantially better name than shuffle for this. The thing is, you need a slightly generic term here because it is a catch all for several different types of memory used at runtime (concerning joins, aggregations, and other buffers). It can be defined in more detail in the tooltip.